core/tenant/project_test.go

```diff
@@ -25,14 +25,14 @@ func TestEntityProject(t *testing.T) {
 
 	t.Run("Project", func(t *testing.T) {
 		t.Run("fails to create if name is empty", func(t *testing.T) {
-			project, err := tenant.NewProject("", map[string]string{"a": "b"})
+			project, err := tenant.NewProject("", map[string]string{"a": "b"}, map[string]string{})
```

Comment:
Can you pass nil here and check if it works with nil.

Reply:
will add another testcase where nil variables param is provided

Reply:
@sbchaos
if given nil, calls to `GetVariables()` will return an empty map (because of the copying behavior) and `GetVariable()` on any key will return error entry not found. No unexpected error or panic occurred